File: 3324-split-the-array/3324-split-the-array.cpp
class Solution {
public:
    bool isPossibleToSplit(vector<int>& nums) {
        unordered_map<int,int>mp; 
        for( auto num : nums){
            if(mp[num] == 2) return false;
            mp[num]++; 
        }
        return true;
        
        
    }
};

File: 3324-split-the-array/3324-split-the-array.java
class Solution {
    public boolean isPossibleToSplit(int[] nums) {
        HashMap<Integer, Integer> mp = new HashMap<>();

        for (int num : nums) {
            int count = mp.getOrDefault(num, 0);
            if (count == 2) return false;
            mp.put(num, count + 1);
        }
        return true;
    }
}
